jgrapht-core/src/main/java/org/jgrapht/alg/spanning/PrimMinimumSpanningTree.java

```diff
@@ -103,6 +103,79 @@ public PrimMinimumSpanningTree(Graph<V, E> graph)
 
         return new SpanningTreeImpl<>(minimumSpanningTreeEdgeSet, spanningTreeWeight);
     }
+
+
+    /**
+     * Computes a spanning tree in $O(|V|^2)$ time.
+     *
+     * Note: This method is only recommended for dense graphs.
+     *
+     * @return a spanning tree
+     */
+    @SuppressWarnings("unchecked")
+    public SpanningTree<E> getSpanningTreeDense(){
+        Set<E> minimumSpanningTreeEdgeSet = new HashSet<>(g.vertexSet().size());
+        double spanningTreeWeight = 0d;
+
+        final int N = g.vertexSet().size();
+
+        /*
+         * Normalize the graph
+         *   map each vertex to an integer (using a HashMap)
+         *   keep the reverse mapping  (using an ArrayList)
+         */
+        Map<V, Integer> vertexMap = new HashMap<>();
+        List<V> indexList = new ArrayList<>();
+        for(V v : g.vertexSet()){
+            vertexMap.put(v,vertexMap.size());
+            indexList.add(v);
+        }
+
+        boolean[] spanned = new boolean[N];
+        double[] distance = new double[N];
+        E[] edgeFromParent = (E[]) new Object[N];
```

Comment:
Create a small inner and private helper class here containing fields spanned, distance and edgeFromParent and use the  safer `Array.newInstance` method for generic array creation. Something along the lines: 

```
private class VertexInfo { 
    public boolean spanned;
    public double distance;
    public E edgeFromParent;
}

VertexInfo[] array = (VertexInfo[]) Array.newInstance(VertexInfo.class, N);
```